hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/RDBStoreAbstractIterator.java

```diff
@@ -150,5 +181,21 @@ public final void removeFromDB() throws IOException {
   @Override
   public void close() {
     rocksDBIterator.close();
+    closed = true;
+    releaseEntry();
+  }
+
+  public static final class AutoCloseableRawKeyValue<RAW> extends RawKeyValue<RAW> implements AutoCloseable {
+    private final UncheckedAutoCloseableSupplier<RawKeyValue<RAW>> keyValue;
+
+    private AutoCloseableRawKeyValue(ReferenceCountedObject<RawKeyValue<RAW>> kv) {
```

Comment:
This itself can be RefrenceCounted AutoClosable class, instead of wrapping at caller.
- retain() --> add reference count
- release()/close() --> release reference count and when "0", close the object.

Reply:
No we should only return the autoCloseable value and not the reference counted object itself if that is what you mean